qiskit/transpiler/passes/layout/vf2_utils.py

```diff
@@ -97,10 +97,13 @@ def _visit(dag, weight, wire_map):
     free_nodes = {}
     if not strict_direction:
         conn_comp = connected_components(im_graph)
-        for comp in conn_comp:
-            if len(comp) == 1:
-                index = comp.pop()
-                free_nodes[index] = im_graph[index]
+    else:
+        conn_comp = weakly_connected_components(im_graph)
```

Comment:
Now we've got an `if not X: ... else: ...` double negation, but we're (hopefully) about to remove this in #14863 anyway, so let's leave it.